src/Straminata.hs

```diff
@@ -82,8 +86,40 @@ player2InitialState = Player
 advanceGame :: Float -- ^ period of time (in seconds) needing to be advanced
             -> Game
             -> Game
--- advanceGame time = updatePlayers . updateCamera . moveObjects time . movePlayers time
-advanceGame time = updateCamera . moveObjects time . movePlayers time
+advanceGame time = updateCamera . moveObjects time . movePlayers time . updateObjects
+
+
+activatePlayer :: String -> Object -> Game -> Game
+activatePlayer name object game =
+  foldr (\player -> (objectOnActivate . playerObject $ player)
+                    (objectsCollide object (playerObject player))
+                    (playerObject player)
```

Comment:
Don't forget that name is unique and you do not really need to iterate over all players.

Reply:
i'm not iterating over all players i'm iterating over 1 player. I like this version better